aws/resource_aws_ecrpublic_repository_test.go

```diff
@@ -0,0 +1,428 @@
+		}
+
+		for _, repository := range page.Repositories {
+			repositoryName := aws.StringValue(repository.RepositoryName)
+			log.Printf("[INFO] Deleting ECR Public repository: %s", repositoryName)
+
+			_, err = conn.DeleteRepository(&ecrpublic.DeleteRepositoryInput{
+				// We should probably sweep repositories even if there are images.
+				Force:          aws.Bool(true),
+				RegistryId:     repository.RegistryId,
+				RepositoryName: repository.RepositoryName,
+			})
+			if err != nil {
+				if !isAWSErr(err, ecrpublic.ErrCodeRepositoryNotFoundException, "") {
+					sweeperErr := fmt.Errorf("Error deleting ECR Public repository (%s): %w", repositoryName, err)
+					log.Printf("[ERROR] %s", sweeperErr)
+					errors = multierror.Append(errors, sweeperErr)
+				}
+				continue
+			}
+		}
+
+		return !isLast
+	})
+	if err != nil {
+		if testSweepSkipSweepError(err) {
+			log.Printf("[WARN] Skipping ECR Public repository sweep for %s: %s", region, err)
+			return nil
+		}
+		errors = multierror.Append(errors, fmt.Errorf("Error retreiving ECR Public repositories: %w", err))
+	}
+
+	return errors
+}
+
+func TestAccAWSEcrPublicRepository_basic(t *testing.T) {
+	var v ecrpublic.Repository
+	rName := acctest.RandomWithPrefix("tf-acc-test")
+	resourceName := "aws_ecrpublic_repository.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSEcrPublicRepositoryDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSEcrPublicRepositoryConfig(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSEcrPublicRepositoryExists(resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "repository_name", rName),
+					testAccCheckAWSEcrPublicRepositoryRegistryID(resourceName),
+					testAccCheckAWSEcrPublicRepositoryRepositoryARN(resourceName, rName),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAWSEcrPublicRepository_catalogdata_abouttext(t *testing.T) {
+	var v ecrpublic.Repository
+	rName := acctest.RandomWithPrefix("tf-acc-test")
+	resourceName := "aws_ecrpublic_repository.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSEcrPublicRepositoryDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSEcrPublicRepositoryCatalogDataConfigAboutText(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSEcrPublicRepositoryExists(resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "catalog_data.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "catalog_data.0.about_text", "About Text"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
```

Comment:
These attribute tests should also verify resource updates, which can be done by parameterizing the test configuration with the attribute value and adding a TestStep after import, e.g.

```suggestion
func TestAccAWSEcrPublicRepository_catalogdata_abouttext(t *testing.T) {
	var v ecrpublic.Repository
	rName := acctest.RandomWithPrefix("tf-acc-test")
	resourceName := "aws_ecrpublic_repository.test"

	resource.ParallelTest(t, resource.TestCase{
		PreCheck:     func() { testAccPreCheck(t) },
		Providers:    testAccProviders,
		CheckDestroy: testAccCheckAWSEcrPublicRepositoryDestroy,
		Steps: []resource.TestStep{
			{
				Config: testAccAWSEcrPublicRepositoryCatalogDataConfigAboutText(rName, "About Text 1"),
				Check: resource.ComposeTestCheckFunc(
					testAccCheckAWSEcrPublicRepositoryExists(resourceName, &v),
					resource.TestCheckResourceAttr(resourceName, "catalog_data.#", "1"),
					resource.TestCheckResourceAttr(resourceName, "catalog_data.0.about_text", "About Text 1"),
				),
			},
			{
				ResourceName:      resourceName,
				ImportState:       true,
				ImportStateVerify: true,
			},
			{
				Config: testAccAWSEcrPublicRepositoryCatalogDataConfigAboutText(rName, "About Text 2"),
				Check: resource.ComposeTestCheckFunc(
					testAccCheckAWSEcrPublicRepositoryExists(resourceName, &v),
					resource.TestCheckResourceAttr(resourceName, "catalog_data.#", "1"),
					resource.TestCheckResourceAttr(resourceName, "catalog_data.0.about_text", "About Text 2"),
				),
			},
		},
	})
}
```

And 

```go
func testAccAWSEcrPublicRepositoryCatalogDataConfigAboutText(rName string, aboutText string) string {
	return fmt.Sprintf(`
resource "aws_ecrpublic_repository" "test" {
  repository_name = %[1]q
  catalog_data {
    about_text = %[2]q
  }
}
`, rName, aboutText)
}
```